### ISSUE TITLE
alfresco-maven-plugin - truezip on 2.0.0

### ISSUE THREAD

**Issue by dhartford** [Alfresco/alfresco-sdk#265]
For the 2.0.0 version of alfresco-maven-plugin, getting these errors:

java.lang.NoClassDefFoundError: de/schlichtherle/truezip/fs/FsSyncExceptionBuilder

some more details here: http://stackoverflow.com/questions/28883051/alfresco-maven-plugin-truezip-issues-to-pre-amp-a-war

using INSTALL alfresco goal

**Comment by gravitonian** [comment]
Hi,

Can you give an example via either the all-in-one project, repo-amp, or share-amp project so it is easier to reproduce your problem. All these projects use the alfresco-maven-plugin to apply amps to the WARs.

**Comment by mindthegab** [comment]
@dhartford was using the wrong version of Maven. See https://stackoverflow.com/questions/28883051/alfresco-maven-plugin-truezip-issues-to-pre-amp-a-war/30489217#30489217 for details.

In 2.1.0 we added a proper maven-enforcer-plugin check (see https://issues.alfresco.com/jira/browse/DEVPLAT-67) to make sure these issues don't happen anymore.

**Comment by Sheldor5** [comment]
I had the same problem with sdk 1.1.1 and I am wondering why this problem wasn't fixed in 2.0.0? Will there ever be a fix for sdk 1.1.1 or does the "we-dont-care-about-old-versions"-argument apply here as well (I am speaking of Alfresco 4.X which is not supported by the sdk 2.X)?

**Comment by mindthegab** [comment]
Enterprise support for the SDK is described here https://www.alfresco.com/services/subscription/technical-support/product-support-status. 1.1.1 is in Limited Support, so it's likely not more work is going in.

But on another note: dude, seriously, instead of wining and complaining, this is a community project, why don't you do a pull request?

I have been running this project for 5 years as a single man show, and now it's getting better since we have a team working on it. The one thing it's not getting better is the community, which I was hoping will start contributing once we moved to Github, but clearly it was not a tool problem.

Lastly: this is a closed issue, with won't fix. Commenting on it won't get you anywhere.

Also, did you bother reading why it was closed? (i.e. @dhartford was using a wrong maven version?)

Did you bother to check your maven version is the correct one for 1.1.1?

One last thing: this issue does not reproduce in standard testing of 1.1.1 so it's likely a problem in *your* environment *you* need to fix, instead of blindly assuming it's an SDK problem.  Just FYI, typically noClassDefFound in Maven is a often quite basic issue, i.e. you have a corrupted JAR in your local machine, so make sure you also try to delete the truezip dep in your local ~/.m2/ maven repo (e.g. http://stackoverflow.com/questions/13159108/maven-throws-error-injecting-and-noclassdeffounderror-when-i-trying-to-build-a)

Ah, and the issue stays closed until you can provide a test case that proves the SDK doesn't work (and maybe provide maven version, java version, exact alfresco version, OS, like any "decent" issue report)?

Thanks!

**Comment by Sheldor5** [comment]
The "Enterprise Support" is neither "Enterprise" nor "Support" so please apologize my wining ... But on the other site I have found many posts about others having this problem and I also tried your suggestions without success. But respect to your -one man show-!

**Comment by mindthegab** [comment]
Hey, apologies if I overreacted. Especially considering your avatar, I should be more respectful @Sheldor5  :)

Not being a one man show anymore (finally!) so that's why I hoped we could be turning around the perception.

What you mean by Enterprise support not being support? Truth to be told we have launched recently an offering just for supporting developers (called Developer Support, http://www.alfresco.com/services/subscription/technical-support/alfresco-developer-support) which is exactly meant to fill the gap between standard support (not geared to support custom development) and what our Customers / Partners really need (i.e. a lot of dev support).

Said that, obviously this is not available for community, unluckily.

But I am happy to try and troubleshoot this. Or make sure that we release 1.2.0 (currently in snapshot https://oss.sonatype.org/#nexus-search;gav~org.alfresco.maven~~1.2.0-SNAPSHOT~~), which is a backport of 2.0.0 to get compatibility with 4.2 with more hot reloading 2.0.0-ish functionalities. 

Let's file a separate issue or try connect with me or @ohej / @gravitonian in the different channels (IRC, mailing list, forums) and I'd be happy to get your feedback / give help.
